langchain_bot.py

```diff
@@ -0,0 +1,66 @@
+from langchain.document_loaders import DirectoryLoader
+
+loader = DirectoryLoader(
+    './Langchain/data/', # my local directory
+    glob='**/*.pdf',     # we only get pdfs
+    show_progress=True
+)
+docs = loader.load()
+
+
+from langchain.text_splitter import CharacterTextSplitter
+
+text_splitter = CharacterTextSplitter(
+    chunk_size=1000, 
+    chunk_overlap=0
+)
+docs_split = text_splitter.split_documents(docs)
+
+import os
+
+PINECONE_API_KEY = ... # find at app.pinecone.io
+PINECONE_ENV = ...     # next to api key in console
+OPENAI_API_KEY = ...   # found at platform.openai.com/account/api-keys
+
+os.environ['OPENAI_API_KEY'] = OPENAI_API_KEY
+
+import pinecone 
+from langchain.vectorstores import Pinecone
+from langchain.embeddings.openai import OpenAIEmbeddings
+
+# we use the openAI embedding model
+embeddings = OpenAIEmbeddings()
+pinecone.init(
+    api_key=PINECONE_API_KEY,
+    environment=PINECONE_ENV
+)
+
+doc_db = Pinecone.from_documents(
+    docs_split, 
+    embeddings, 
+    index_name='langchain-demo'
+)
+
+
+query = "What were the most important events for Google in 2021?"
+search_docs = doc_db.similarity_search(query)
+print(search_docs)
+
+from langchain import OpenAI
+llm = OpenAI()
+
+from langchain.chat_models import ChatOpenAI
+llm = ChatOpenAI()
+
+from langchain.chains import RetrievalQA
+
+qa = RetrievalQA.from_chain_type(
+    llm=llm, 
+    chain_type='stuff',
+    retriever=doc_db.as_retriever(),
+)
+
+query = "What were the earnings in 2022?"
+result = qa.run(query)
+
+print(result)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The queries are hardcoded which limits the flexibility of the bot. Consider making the bot more interactive by allowing users to input their own queries. Also, the `llm` variable is redefined on line 53, which might be an error. Ensure that the correct model is being used.
```diff
- query = "What were the most important events for Google in 2021?"
- search_docs = doc_db.similarity_search(query)
- print(search_docs)

- ...

- query = "What were the earnings in 2022?"
- result = qa.run(query)
- print(result)

+ while True:
+     query = input("Enter your query: ")
+     if query.lower() == 'quit':
+         break
+     result = qa.run(query)
+     print(result)

- llm = OpenAI()
- ...
- llm = ChatOpenAI()

+ llm = OpenAI() if some_condition else ChatOpenAI()
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->